internal/app/machined/pkg/runtime/v1alpha1/platform/azure/azure.go

```diff
@@ -241,9 +275,30 @@ func (a *Azure) NetworkConfiguration(ctx context.Context, ch chan<- *runtime.Pla
 
 	networkConfig, err := a.ParseMetadata(interfaceAddresses, host)
 	if err != nil {
+		return fmt.Errorf("failed to parse network metadata: %w", err)
+	}
+
+	log.Printf("fetching load balancer metadata from: %q", AzureHostnameEndpoint)
+
+	var loadBalancerAddresses LoadBalancerMetadata
+
+	lbConfig, err := download.Download(ctx, AzureLoadbalancerEndpoint,
+		download.WithHeaders(map[string]string{"Metadata": "true"}),
+		download.WithErrorOnNotFound(errors.ErrNoHostname),
+		download.WithErrorOnEmptyResponse(errors.ErrNoHostname))
+	if err != nil && !stderrors.Is(err, errors.ErrNoHostname) {
 		return err
 	}
 
+	if err = json.Unmarshal(lbConfig, &loadBalancerAddresses); err != nil {
```

Comment:
if we fail to download, should we skip the unmarshal?

Reply:
Yep, I added more details on error.